include/zephyr/logging/log.h

```diff
@@ -328,8 +339,7 @@ void z_log_vprintk(const char *fmt, va_list ap);
 		log_source_const_data,								\
 		Z_LOG_ITEM_CONST_DATA(_name)) =							\
 	{											\
-		.name = IS_ENABLED(CONFIG_LOG_FMT_SECTION_STRIP) ? NULL :			\
-			COND_CODE_1(CONFIG_LOG_FMT_SECTION,					\
+		.name = COND_CODE_1(LOG_MODULE_NAME_IN_SECTION,					\
```

Comment:
So, now there are actually cases where the name is pointing to invalid flash again. Isn't it safer to assign it to NULL here instead of in log_source_name_get. Otherwise you need to rely on everyone accessing name through that function.

If not then do yourself a favour an update here: https://github.com/nrfconnect/sdk-nrf/blob/main/samples/matter/common/src/diagnostic/log_backend_diagnostic.cpp#L52

Reply:
I did not want to assign null because then it would not be used and removed from elf. I did not notice that `__used` is used for module name variable so this pr makes no sense.